addons/mrp/migrations/13.0.2.0/post-migration.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2020 Payam Yasaie <https://www.tashilgostar.com>
+# Copyright 2020 Andrii Skrypka <andrijskrypa@ukr.net>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_unlink_by_xmlid = [
+    'mrp.sequence_mrp_unbuild',
+]
+
+
+def convert_many2one_field(env):
+    openupgrade.m2o_to_x2m(
+        env.cr,
+        env['stock.move.line'], 'stock_move_line',
+        'lot_produced_ids', openupgrade.get_legacy_name('lot_produced_id')
+    )
+
+
+def fill_unbuild_company_id(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE mrp_unbuild mu
+        SET company_id = COALESCE(
+            (SELECT sm.company_id FROM stock_move sm
+             WHERE mu.id = sm.unbuild_id AND sm.company_id IS NOT NULL
+             LIMIT 1), ru.company_id)
+        FROM res_users ru
+        WHERE ru.id = mu.create_uid AND mu.company_id IS NULL
+        """
+    )
+
+
+def fill_stock_picking_type_sequence_code(env):
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE stock_picking_type spt
+        SET sequence_code = CASE
+          WHEN spt.id = wh.pbm_type_id THEN 'PC'
+          WHEN spt.id = wh.sam_type_id THEN 'SFP'
+          WHEN spt.id = wh.manu_type_id THEN 'MO'
+          END
+        FROM stock_warehouse wh
+        WHERE sequence_code = 'TO_FILL' AND spt.id in (wh.pbm_type_id, wh.sam_type_id, wh.manu_type_id)
+        """
+    )
+
+
+def handle_unbuild_sequence(env):
+    # although later the 'mrp.sequence_mrp_unbuild' sequence will be deleted,
+    # we need to nullify its code (if having one) here
+    # because we want a new autogenerated sequence
+    openupgrade.logged_query(
+        env.cr, """
+        UPDATE ir_sequence seq
+        SET code = NULL
+        FROM ir_model_data imd
+        WHERE imd.res_id = seq.id AND imd.module = 'mrp'
+            AND imd.name = 'sequence_mrp_unbuild'"""
+    )
+    # force execute this function (it is noupdate=1 in xml data)
+    env['res.company'].create_missing_unbuild_sequences()
+
+
+def fill_manufacture_mto_pull(env):
+    warehouses = env['stock.warehouse'].search([
+        ('manufacture_to_resupply', '=', True),
+        ('manufacture_mto_pull_id', '=', False),
+    ])
+    for wh in warehouses:
+        rule_details = wh._get_global_route_rules_values()['manufacture_mto_pull_id']
+        values = rule_details.get('update_values', {})
+        values.update(rule_details['create_values'])
+        values.update({'warehouse_id': wh.id})
+        wh['manufacture_mto_pull_id'] = env['stock.rule'].create(values).id
+
+
+def fill_mrp_workorder_product_uom_id(cr):
+    openupgrade.logged_query(
+        cr, """
+        UPDATE mrp_workorder mw
+        SET product_uom_id = mp.product_uom_id
+        FROM mrp_production mp
+        WHERE mw.production_id = mp.id AND mw.product_uom_id IS NULL

```

Comment:
If you want to read `uom_id` field directly from `product_template` table, you can act as follows:
```sql
UPDATE mrp_workorder w
SET    product_uom_id = pt.uom_id
FROM   product_product pp
JOIN   product_template pt ON pp.product_tmpl_id = pt.id
WHERE  w.product_id = pp.id
AND    w.product_uom_id IS NULL;
```

Reply:
@dennybiasiolli 
I don't know why you thought that.
I set product uom according to odoo12
See: https://github.com/odoo/odoo/blob/3ca2a1903da2431bbf72d6112d659d79d470f249/addons/mrp/models/mrp_workorder.py#L39

Reply:
You're right, I think it was "cleaner" to read `product_uom_id` directly from `product_template`, but looking at the link you provided it's cleaner your way ;)